apis/clusters/v1beta1/kafka_webhook_test.go

```diff
@@ -0,0 +1,227 @@
+package v1beta1
+
+import (
+	"context"
+	"os"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/yaml"
+
+	"github.com/instaclustr/operator/pkg/models"
+)
+
+var _ = Describe("Kafka Controller", Ordered, func() {
+	kafkaManifest := Kafka{}
+	kafkaNamespacedName := &types.NamespacedName{}
+	TestKafkaManifest := Kafka{}
+
+	It("Reading kafka manifest", func() {
+		yfile, err := os.ReadFile("../../../controllers/clusters/datatest/kafka_v1beta1.yaml")
+		Expect(err).Should(Succeed())
+
+		err = yaml.Unmarshal(yfile, &kafkaManifest)
+		Expect(err).Should(Succeed())
+		kafkaNamespacedName.Name = kafkaManifest.ObjectMeta.Name
+		kafkaNamespacedName.Namespace = defaultNS
+		TestKafkaManifest = kafkaManifest
+	})
+
+	ctx := context.Background()
+
+	When("apply a Kafka manifest", func() {
+		It("should test kafka creation flow", func() {
+
+			TestKafkaManifest.Spec.Description = "some description"
+			Expect(k8sClient.Create(ctx, &TestKafkaManifest)).ShouldNot(Succeed())
+			TestKafkaManifest.Spec.Description = kafkaManifest.Spec.Description
+
+			if len(TestKafkaManifest.Spec.TwoFactorDelete) == 0 {
+				TestKafkaManifest.Spec.TwoFactorDelete = []*TwoFactorDelete{{Email: "test@test.com"}, {Email: "test@test.com"}}
+			} else {
+				TestKafkaManifest.Spec.TwoFactorDelete = []*TwoFactorDelete{kafkaManifest.Spec.TwoFactorDelete[0], {Email: "test@test.com"}}
+			}
+			Expect(k8sClient.Create(ctx, &TestKafkaManifest)).ShouldNot(Succeed())
+			TestKafkaManifest.Spec.TwoFactorDelete = kafkaManifest.Spec.TwoFactorDelete
+
+			TestKafkaManifest.Spec.SLATier = "some SLATier that is not support"
```

Comment:
some SLATier that is not support -> some SLATier that is not supported

Reply:
agree, changed